src/lib/hooks/swap/useWrapCallback.tsx

```diff
@@ -118,7 +111,11 @@ export default function useWrapCallback(): UseWrapCallbackReturns {
     setWrapState((state) => ({ ...state, loading: true }))
     const result = await (wrapType === WrapType.WRAP
       ? wrappedNativeCurrencyContract.deposit({ value: `0x${parsedAmountIn.quotient.toString(16)}` })
-      : wrappedNativeCurrencyContract.withdraw(`0x${parsedAmountIn.quotient.toString(16)}`))
+      : wrappedNativeCurrencyContract.withdraw(`0x${parsedAmountIn.quotient.toString(16)}`)
+    ).catch(() => {
+      setWrapState((state) => ({ ...state, loading: false }))
+      return Promise.reject('User denied wallet transaction.')
```

Comment:
This may fail for other reasons. Is there a reason you can't just have the actual error returned?

What about replacing the catch with a `.finally(() => setWrapState((state) => ({ ...state, loading: false })))` to deal with resetting the loading state?

Reply:
we do this on line 120, but if the request to wallet fails then the `finally` doesnt get triggered. I think this is probably due to the `wait(1)` where its waiting for block confirmation (which doesnt happen if error occurs when sent to wallet)

Reply:
we could replace this catch with a finally though, which may be more correct (because it can fail for other reasons)

Reply:
I see. Could you keep it as a catch, but rethrow?

```
).catch((e: unknown) => {
  setWrapState(...)
  throw e
})
```

Reply:
I like this, will update with that